components/tallinn-experience/TallinnInterestForm.tsx

```
@@ -0,0 +1,356 @@
+}
+
+const intentLabel: Record<(typeof TALLINN_ATTENDANCE_INTENTS)[number], string> = {
+  exploring: 'Exploring',
+  likely: 'Likely if the format fits',
+  'ready-if-time-works': 'Ready if one time works',
+}
+
+function safeVariant() {
+  const params = new URLSearchParams(window.location.search)
+  const candidate = params.get('variant') || params.get('ref') || 'default'
+  return /^[a-z0-9-]{1,60}$/.test(candidate) ? candidate : 'default'
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)